artemis-core-client/src/main/java/org/apache/activemq/artemis/core/client/impl/ClientSessionFactoryImpl.java

```diff
@@ -90,7 +90,7 @@
 
    private ConnectorFactory connectorFactory;
 
-   private transient boolean finalizeCheck = true;
```

Comment:
this will clash with the other PR.. you will have to remove this change after I merge the other.